src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,171 @@
+#include "PlayfairCipher.hpp"
+
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <iterator>
+
+PlayfairCipher::PlayfairCipher(const std::string& key){
+    setKey(key);
+}
+
+std::string encounteredletters{""};
+
+void PlayfairCipher::setKey(const std::string& key){
+    // store the original key
+    key_ = key;
+
+    // Append the alphabet
+    key_.append("abcdefghijklmnopqrstuvwxyz");
+
+    // Make sure the key is upper case
+    
+    std::transform(key_.begin(),key_.end(),key_.begin(),toupper);   //transform algorithm which transforms all char in string to upper case
+
+    // Remove non-alpha characters
+    auto iter = std::remove_if(key_.begin(),key_.end(),[](char& val){return !std::isalpha(val);});
+    key_.erase(iter,key_.end());
+    
+    // Change J -> I
+    std::transform(key_.begin(),key_.end(),key_.begin(),[](char& val){if(val=='J'){return 'I';} else {return val;}});
+    
+    // Remove duplicated letters
+    auto iter2 = std::remove_if(key_.begin(),key_.end(),[=](char& val){
+        if(encounteredletters.find(val)!=std::string::npos){
+            return true;
+        } else {
+            encounteredletters += val; 
+            return false;
+        }});
+    key_.erase(iter2,key_.end());
+
+    // Store the coords of each letter and the playfair cipher key map
+
+    for(std::size_t i{0};i<key_.size();i++){
+        Str2CoordMap::value_type invcoords{std::pair<int,int> {(i/5)+1,(i%5)+1},key_[i]};
+        Str2CharMap::value_type coords{key_[i],std::pair<int,int> {(i/5)+1,(i%5)+1}};
+
+        playfairciphermap_.insert(coords);
+        playfairciphermapinv_.insert(invcoords);
+    }
+    /*
+    for ( auto p : playfairciphermap_ ){
+        std::cout << p.first << ": " << p.second.first << ":" << p.second.second << std::endl;
+    }
+
+    for (auto p : playfairciphermapinv_){
+        std::cout << p.first.first << ":" << p.first.second << " :" << p.second << std::endl;
+    }
+    */
+    
+}
+
+
+std::string PlayfairCipher::applyCipher(const std::string& inputText,
+                                        const CipherMode cipherMode) const {
+
+    std::string output{""},modifiedinput{""},input{inputText};
+
+    switch(cipherMode){
+        case CipherMode::Encrypt:
+            // Change J → I
+            std::transform(input.begin(),input.end(),input.begin(),[](char& val){if(val=='J'){return 'I';} else {return val;}});
+            std::cout << "after j->i: " << input << std::endl;
```

Comment:
Debug statements in development like this are absolutely fine (and required!) but try to remove them or at least comment them out when pushing to a production repository.